### PR TITLE
nlopt install to ros/hydro/devel/lib and ros/hydro/build dir

### PR DESCRIPTION
**Pull request by s-noda**

### CODE REVIEW

File: 3rdparty/nlopt/CMakeLists.txt [hunk 1]
```diff
@@ -5,11 +5,11 @@ cmake_minimum_required(VERSION 2.4.6)
 find_package(catkin REQUIRED COMPONENTS mk)
 
 add_custom_target(libnlopt_cxx ALL
-  DEPENDS ${PROJECT_SOURCE_DIR}/lib/libnlopt_cxx.so)
-add_custom_command(OUTPUT ${PROJECT_SOURCE_DIR}/lib/libnlopt_cxx.so
-  COMMAND cmake -E chdir ${PROJECT_SOURCE_DIR} make -f ${PROJECT_SOURCE_DIR}/Makefile DSTDIR=${PROJECT_SOURCE_DIR}
+  DEPENDS ${CATKIN_DEVEL_PREFIX}/lib/libnlopt_cxx.so)
+add_custom_command(OUTPUT ${CATKIN_DEVEL_PREFIX}/lib/libnlopt_cxx.so
+  COMMAND cp ${PROJECT_SOURCE_DIR}/nlopt-2.3.tar.gz.md5sum ${CMAKE_CURRENT_BINARY_DIR}
```
Comment (garaemon) [inline 3rdparty/nlopt/CMakeLists.txt]: Can you download sourcecode into build space?

Comment (s-noda) [inline 3rdparty/nlopt/CMakeLists.txt]: Yes, nlopt source code will be downloaded into ros/hydro/build... .

Comment (garaemon) [inline 3rdparty/nlopt/CMakeLists.txt]: I see, only md5sum is located under source directory

**Comment by garaemon** [comment]
Have you confirmed that eus_nlopt can be compiled?

**Comment by garaemon** [comment]
```
[[36mnlopt[0m] [100%] Built target libnlopt_cxx[0m [0m[0m
[[36mnlopt[0m] [36mInstall the project...[0m [0m[0m
[[36mnlopt[0m] [0m-- Install configuration: ""[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/_setup_util.py[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/env.sh[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/setup.bash[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/setup.sh[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/setup.zsh[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/.rosinstall[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/etc/catkin/profile.d/05.catkin_make.bash[0m [0m[0m
[[36mnlopt[0m] -- Up-to-date: /home/travis/ros/ws_jsk_common/install/etc/catkin/profile.d/05.catkin_make_isolated.bash[0m [0m[0m
[[36mnlopt[0m] -- Installing: /home/travis/ros/ws_jsk_common/install/lib/pkgconfig/nlopt.pc[0m [0m[0m
[[36mnlopt[0m] -- Installing: /home/travis/ros/ws_jsk_common/install/share/nlopt/cmake/nloptConfig.cmake[0m [0m[0m
[[36mnlopt[0m] -- Installing: /home/travis/ros/ws_jsk_common/install/share/nlopt/cmake/nloptConfig-version.cmake[0m [0m[0m
[[36mnlopt[0m] -- Installing: /home/travis/ros/ws_jsk_common/install/share/nlopt/package.xml[0m [0m[0m
[[36mnlopt[0m] CMake Error at cmake_install.cmake:105 (FILE):[0m [0m[0m
[[36mnlopt[0m]   file INSTALL cannot find[0m [0m[0m
[[36mnlopt[0m]   "/home/travis/ros/ws_jsk_common/src/jsk_common/3rdparty/nlopt/lib".[0m [0m[0m
[[36mnlopt[0m] [0m [0m[0m
[[36mnlopt[0m] [0m [0m[0m
[[36mnlopt[0m] make: *** [install] Error 1[0m [0m[0m

```

**Comment by s-noda** [comment]
Yes, I confirmed using my laptop PC and the other @mmurooka 's desktop PC.

**Comment by garaemon** [comment]
> Yes, I confirmed using my laptop PC and the other @mmurooka 's desktop PC.

great

**Comment by garaemon** [comment]
Please do not remove install function

**Comment by garaemon** [comment]
Installing library is required for eus_nlopt

**Comment by garaemon** [comment]
great :smile:

**Pull request merged**